### DISCUSSION TITLE
Custom settings or lambdas with rule parameters

### DISCUSSION THREAD

**Discussion by megyltv** [microsoft/RulesEngine#154] [Q&A]
1. I'm using rule parameters and when I evaluate using a custom setting (ReSetting), the value isn't evaluate. (Expression: "Utils.MaximumCharacters(input1.value, 4) == true") 
2. In other scenario the following lambda expression is defined: "string.IsNullOrEmpty(input1.value) == true", but with rules parameters isn't evaluated correctly.

Both scenarios evaluates correctly when I'm using "ExecuteAllRulesAsync(string name, object[) inputs" method.

Is it necessary to add something before executing the rules in both scenarios?

**Comment by abbasc52** [comment]
Hi @megyltv, could you provide me the code sample to reproduce it?
Also can you provide more details on ` but with rules parameters isn't evaluated correctly.` part?
